lib/instrumentation/modules/cassandra-driver.js

```diff
@@ -0,0 +1,160 @@
+'use strict'
+
+const semver = require('semver')
+const sqlSummary = require('sql-summary')
+
+const shimmer = require('../shimmer')
+
+module.exports = function (cassandra, agent, version, enabled) {
+  if (!enabled) return cassandra
+  if (!semver.satisfies(version, '^3.0.0')) {
+    agent.logger.debug('cassandra-driver version %s not supported - aborting...', version)
+    return cassandra
+  }
+
+  if (cassandra.Client) {
+    shimmer.wrap(cassandra.Client.prototype, 'connect', wrapConnect)
+    shimmer.wrap(cassandra.Client.prototype, 'execute', wrapExecute)
+    shimmer.wrap(cassandra.Client.prototype, 'eachRow', wrapEachRow)
+    shimmer.wrap(cassandra.Client.prototype, 'batch', wrapBatch)
+  }
+
+  return cassandra
+
+  function wrapConnect (original) {
+    return function wrappedConnect (callback) {
+      const span = agent.buildSpan()
+      if (!span) {
+        return original.apply(this, arguments)
+      }
+
+      span.type = 'db.cassandra.connect'
+      span.start()
+
+      function resolve () {
+        span.end()
+      }
+
+      // Wrap the callback
+      const ret = original.call(this, wrapCallback(callback))
+
+      if (typeof callback !== 'function') {
+        ret.then(resolve, resolve)
+      }
+
+      return ret
+
+      function wrapCallback (cb) {
+        if (!cb) return
```

Comment:
It might be safest to expand this check to `typeof cb !== 'function'`

Reply:
Also, it should probably then also return the callback, so it's as if you called `original` with that argument just as if we hadn't patched it.

Reply:
fixed.